src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs

```diff
@@ -233,4 +233,20 @@ public static int GetReplicaCount(this IResource resource)
             return 1;
         }
     }
+
+    /// <summary>
+    /// Gets the lifetime type of the container for the specified resoruce. Defaults to <see cref="ContainerLifetimeType.AppHost"/> if
+    /// no <see cref="ContainerLifetimeAnnotation"/> is found.
+    /// </summary>
+    /// <param name="resource">The resource to the get the ContainerLifetimeType for.</param>
+    /// <returns></returns>
+    public static ContainerLifetimeType GetContainerLifetimeType(this IResource resource)
```

Comment:
This is the only thing that gives me pause right now. I'm wondering whether it might be best to make this internal for now. We probably want to mark the WithContainerLifetime API as experimental.

Reply:
Yeah, I felt a bit odd about the method since it's only really relevant to a resource that also has a container image annotation, but it also simplifies the ApplicationExecutor logic quite a bit.